src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let finalString = '';
+
+    if ((this[0] !== null)
+        && (this[0] !== undefined)) {
+      finalString += this[0];
+    }
```

Comment:
```suggestion
```
looks like redundant if.
you can start from 0 in loop
and check i !== 0 for add separator